src/Diff.ml

```diff
@@ -0,0 +1,107 @@
+let redPixel = (255, 0, 0)
+let maxYIQPossibleDelta = 35215.
+
+type 'a diffVariant = Layout | Pixel of ('a * int * float * int Stack.t)
```

Comment:
The `(...)` here causes an extra tuple to be allocated here, is that intended? If we leave that out then each component directly becomes a member of the data constructor: `Pixel of 'a * int * float * int Stack.t`.

But in exchange you can't separately have a tuple and then wrap it in `Pixel ...`, you have to create the `Pixel (...)` value in one shot.

Reply:
It is not allocated, it is a return type of the other function